app/repositories/beatmap_mirror_requests.py

```diff
@@ -49,9 +49,9 @@ async def get_mirror_weight(mirror_name: str) -> int:
     if failure_rate is None:
         return MIRROR_INITIAL_WEIGHT
 
-    # https://www.desmos.com/calculator/0am8xnwxyo
-    latency_weight = 1000 / math.log(p90_success_ms_latency)
-    failure_weight = math.exp(-10 * failure_rate)
+    # https://www.desmos.com/calculator/wxpsjhdby9
+    latency_weight = 1000 * math.exp(-1 / 1000 * p90_success_ms_latency)
```

Comment:
I felt that the previous calculation was not having a great enough affect between the (realistic) values of ~500 - 3000 ms.

The exponential part of the function was having a great effect between 0ms and 400ms, however in practice those numbers are practically never observed. Conversely, the function was not having a great enough effect in penalizing the difference between say, 1000ms and 4000ms, which was only dropping the rating from ~330 to 270.

Now in the updated formula, both of these concerns are resolved:

[!image](https://i.cmyui.xyz/4YpExBsKRlMPI9w.png)

(Old function in blue, new function in black)